src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/WorkspaceProjectStateChangeDetector.cs

```diff
@@ -292,23 +302,27 @@ private async Task Workspace_WorkspaceChangedAsync(WorkspaceChangeEventArgs e, C
                 Debug.Fail("WorkspaceProjectStateChangeDetector.Workspace_WorkspaceChanged threw exception:" +
                     Environment.NewLine + ex.Message + Environment.NewLine + "Stack trace:" + Environment.NewLine + ex.StackTrace);
             }
+        }
 
-            static bool IsRazorFileOrRazorVirtual(Document document)
-            {
-                // Using EndsWith because Path.GetExtension will ignore everything before .cs
-                // Using Ordinal because the SDK generates these filenames.
-                // Stll have .cshtml.g.cs and .razor.g.cs for Razor.VSCode scenarios.
-                return document.FilePath.EndsWith(".cshtml.g.cs", FilePathComparison.Instance) ||
-                    document.FilePath.EndsWith(".razor.g.cs", FilePathComparison.Instance) ||
-                    document.FilePath.EndsWith(".razor", FilePathComparison.Instance) ||
+        private bool IsRazorFileOrRazorVirtual(Document document)
+        {
+            if (document.FilePath is null)
+                return false;
 
-                    // VSCode's background C# document
-                    document.FilePath.EndsWith("__bg__virtual.cs", StringComparison.Ordinal);
-            }
+            // Using EndsWith because Path.GetExtension will ignore everything before .cs
+            return document.FilePath.EndsWith(_languageServerFeatureOptions.CSharpVirtualDocumentSuffix, FilePathComparison.Instance) ||
+                // Still have .cshtml.g.cs and .razor.g.cs for Razor.VSCode scenarios.
+                document.FilePath.EndsWith(".cshtml.g.cs", FilePathComparison.Instance) ||
+                document.FilePath.EndsWith(".razor.g.cs", FilePathComparison.Instance) ||
+                document.FilePath.EndsWith(".razor", FilePathComparison.Instance) ||
+
+                // VSCode's background C# document
+                // Using Ordinal because the SDK generates these filenames.
+                document.FilePath.EndsWith("__bg__virtual.cs", StringComparison.Ordinal);
```

Comment:
Hmm, I wonder when VSCode generates `__bg__virtual.cs` vs. `.razor.g.s`/`.cshtml.g.cs`?

Reply:
No idea! I definitely took a cautious approach with this, as I doubt these paths will come up in normal circumstances :)